# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1548514289 0
#      Sat Jan 26 14:51:29 2019 +0000
# Node ID cf4792ae08a14d519ae5ac2c9013e5b5d91f7fd9
# Parent  41e2de681481b99e5f08a2af12c770027b8776e2
* pt-misc.cc (tree_parameter_list::dup): Always copy m_marked_for_varargs.

diff --git a/libinterp/parse-tree/pt-misc.cc b/libinterp/parse-tree/pt-misc.cc
--- a/libinterp/parse-tree/pt-misc.cc
+++ b/libinterp/parse-tree/pt-misc.cc
@@ -59,18 +59,17 @@ namespace octave
     return retval;
   }
 
   tree_parameter_list *
   tree_parameter_list::dup (symbol_scope& scope) const
   {
     tree_parameter_list *new_list = new tree_parameter_list ();
 
-    if (takes_varargs ())
-      new_list->mark_varargs ();
+    new_list->m_marked_for_varargs = m_marked_for_varargs;
 
     for (const tree_decl_elt *elt : *this)
       new_list->append (elt->dup (scope));
 
     return new_list;
   }
 
   // Return lists.
